test/sql/join/iejoin/iejoin_issue_10122.test

```diff
@@ -0,0 +1,40 @@
+# name: test/sql/join/iejoin/iejoin_issue_10122.test
```

Comment:
Can we rename this test file to test_iejoin_null_keys? I know some of the others are named with issue numbers but that is really not very descriptive and I apologise for the chaos.